Use picojson bundled with mesos

branden opened the pull request:
This allows the isolator's configure script to locate `picojson.h` when run with `--with-mesos` or `--with-mesos-build-dir`.

See https://github.com/apache/mesos/commit/142157dc787538a6ef59678c6ac6c337ff743ab2.

branden committed:
Use picojson bundled with mesos

This allows the isolator's configure script to locate `picojson.h` when
run with `--with-mesos` or `--with-mesos-build-dir`.

cantbewong commented:
could you sync up with current base branch and resubmit - then I will merge

In oarticular Dockerfile-mesos-modules-dev was changed for 0.27.1 so the deletion of the wget for picojson should be applied to that file

branden commented:
I rebased onto current master. I also added a fix that allows `--with-mesos-build-dir` to find the bundled picojson, so please give the commit another look. Thanks!
